### PR TITLE
Add model EMA (Exponential Moving Average)

### PR DESCRIPTION
**Pull request by sokovninn**
This pull request introduces a new Exponential Moving Average (EMA) callback for model training, along with its integration and corresponding unit tests. The main changes include the addition of the `EMACallback` class, its registration, and tests to ensure its functionality.

### New EMA Callback

* [`luxonis_train/callbacks/ema.py`](diffhunk://#diff-e73f9f2330d6d2ca1a016ad0d897f901c38cf58e9a33d905374798ff41d2a301R1-R227): Added the `EMACallback` class, which includes methods for initializing the EMA, updating EMA weights, and handling checkpoint saving/loading. This class helps in maintaining a moving average of model parameters to potentially improve model performance.

### Integration

* [`luxonis_train/callbacks/__init__.py`](diffhunk://#diff-9fb27879fda34ab85599e83a085fe46f99a42acbaeea39b48c714be901cce3ebR16): Registered the `EMACallback` in the `CALLBACKS` registry and added it to the `__all__` list for proper module export. [[1]](diffhunk://#diff-9fb27879fda34ab85599e83a085fe46f99a42acbaeea39b48c714be901cce3ebR16) [[2]](diffhunk://#diff-9fb27879fda34ab85599e83a085fe46f99a42acbaeea39b48c714be901cce3ebR38) [[3]](diffhunk://#diff-9fb27879fda34ab85599e83a085fe46f99a42acbaeea39b48c714be901cce3ebR52)

### Unit Tests

* [`tests/unittests/test_callbacks/test_ema.py`](diffhunk://#diff-1059ecbad0484c2a7851bf6780e22dbbed5b04955b1b1c687d1da8066b6a54eaR1-R94): Added unit tests for the `EMACallback` to verify its initialization, batch update, checkpoint saving/loading, and validation epoch handling.

### CODE REVIEW

**Review by klemen1999 — COMMENTED** [review]
Overall LGTM, let's also add it do the docs (callback README)

File: luxonis_train/callbacks/ema.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    Keeps a moving average of everything in the model.state_dict
+    (parameters and buffers).
+    """
+
+    def __init__(
+        self,
+        model: pl.LightningModule,
+        decay: float = 0.9999,
+        use_dynamic_decay: bool = True,
+        device: str = None,
```
[the remark this answers is not in the document]
Comment (sokovninn) [reply luxonis_train/callbacks/ema.py:27]: You may want to keep the average weights on the CPU to reduce GPU memory usage.

[the remark this answers is not in the document]
Comment (klemen1999) [reply luxonis_train/callbacks/ema.py:27]: I see. But does it work also if you don't specify it explicitly? In that case does it just take the device of the model?

[the remark this answers is not in the document]
Comment (sokovninn) [reply luxonis_train/callbacks/ema.py:27]: Yes, if the device is not specified, it defaults to the device of the original model

**Review by kozlov721 — APPROVED** [review]
LGTM

**Pull request merged**